pkg/config/node/minimum_kubelet_version.go

```diff
@@ -0,0 +1,80 @@
+package node
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/blang/semver/v4"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+// An error to be returned when kubelet version is lower than specified minimum kubelet version.
+// Used to differentiate between a parsing failure, and a failure because the kubelet is out of date.
+var ErrKubeletOutdated = fmt.Errorf("kubelet version is outdated")
+
+// ValidateMinimumKubeletVersion takes a list of nodes and a currently set min version.
+// It parses the min version and iterates through the nodes, comparing the version of the kubelets
+// to the min version.
+// It will error if any nodes are older than the min version.
+func ValidateMinimumKubeletVersion(nodes []*corev1.Node, minimumKubeletVersion string) error {
+	// unset, no error
+	if minimumKubeletVersion == "" {
+		return nil
+	}
+
+	version, err := semver.Parse(minimumKubeletVersion)
+	if err != nil {
+		return fmt.Errorf("Failed to parse submitted version %s %v", minimumKubeletVersion, err.Error())
+	}
+
+	for _, node := range nodes {
+		if err := IsNodeTooOld(node, &version); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// IsNodeTooOld answers that very question. It takes a node object and a minVersion,
+// parses each into a semver version, and then determines whether the version of the kubelet on the
+// node is older than min version.
+// When the node is too old, it returns the error ErrKubeletOutdated. If a different error occurs, an error is returned.
+// If the node is new enough and no error happens, nil is returned.
+func IsNodeTooOld(node *corev1.Node, minVersion *semver.Version) error {
```

Comment:
Do the other functions need to be exported?

Reply:
we'll use them elsewhere in [KAS](https://github.com/openshift/kubernetes/pull/2104) and [hypershift](https://github.com/openshift/hypershift/pull/4980)